In [115]:
#Esto unicamente es para ver el texto del doc aqui
# Abrimos  el txt en formato legible
with open ("Calidad de agua 2021.txt", "r", encoding = "utf-8") as f:
    contenido_texto = f.read()

# Esto es un adicional para que no haya una linea gigante

# Reemplazar cada punto con un punto seguido de un salto de línea
contenido_formateado = contenido_texto.replace('. ', '.\n')

## en otro momento mirar si se puede meter un salto de linea cada x palabras o algo asi

print(contenido_formateado)

CALIDAD DE AGUA DEL RÍO DE LA PLATA - 2021
Sobre el conjunto de datos
Este conjunto de datos incluye 166 mediciones en diferentes sitios del Río de La Plata durante 2021, y en general, uno en cada estación del año.

Descripción de las variables
1.
sitios: Localización específica donde se realizó el muestreo del agua.
2.
codigo: Identificador único para cada muestra o estación de muestreo.
3.
fecha: Fecha en la que se tomó la muestra de agua.
4.
año: Año en que se realizó el muestreo.
5.
campaña: Nombre o número de la campaña de monitoreo en la que se realizó el muestreo.
6.
tem agua: Temperatura del agua en grados Celsius.
7.
tem aire: Temperatura del aire en grados Celsius.
8.
od: Oxígeno disuelto, medido en miligramos por litro (mg/L), esencial para la vida acuática.
9.
ph: Medida de la acidez o alcalinidad del agua, en una escala de 0 a 14.
10.
olores: Presencia de olores en el agua, que puede indicar contaminación.
11.
color: Color del agua, que puede ser un indicador de la calidad

COMO NO SE LEE UNA MIERDA PORQ EL PYTHON ... entonces copio y pego aca

CALIDAD DE AGUA DEL RÍO DE LA PLATA - 2021
     
Sobre el conjunto de datos
Este conjunto de datos incluye 166 mediciones en diferentes sitios del Río de La Plata durante 2021, 
y en general, uno en cada estación del año. 
Descripción de las variables
1. sitios: Localización específica donde se realizó el muestreo del agua.
2. codigo: Identificador único para cada muestra o estación de muestreo.
3. fecha: Fecha en la que se tomó la muestra de agua.
4. año: Año en que se realizó el muestreo.
5. campaña: Nombre o número de la campaña de monitoreo en la que se realizó el muestreo.
6. tem agua: Temperatura del agua en grados Celsius.
7. tem aire: Temperatura del aire en grados Celsius.
8. od: Oxígeno disuelto, medido en miligramos por litro (mg/L), esencial para la vida acuática.
9. ph: Medida de la acidez o alcalinidad del agua, en una escala de 0 a 14.
10. olores: Presencia de olores en el agua, que puede indicar contaminación.
11. color: Color del agua, que puede ser un indicador de la calidad del agua.
12. espumas: Presencia de espumas en la superficie del agua, que puede ser un signo de contaminación.
13. mat susp: Materia suspendida, que se refiere a partículas sólidas que flotan en el agua.
14. colif fecales ufc 100ml: Unidades formadoras de colonias de coliformes fecales en 100 ml de agua, un indicador de contaminación fecal.
15. escher coli ufc 100ml: Unidades formadoras de colonias de Escherichia coli en 100 ml de agua, otro indicador de contaminación fecal.
16. enteroc ufc 100ml: Unidades formadoras de colonias de enterococos en 100 ml de agua, que también indican contaminación fecal.
17. nitrato mg l: Concentración de nitratos en miligramos por litro (mg/L), que puede indicar contaminación por fertilizantes.
18. nh4 mg l: Concentración de amonio en miligramos por litro (mg/L), que puede ser un indicador de contaminación orgánica.
19. p total l mg l: Fósforo total en miligramos por litro (mg/L), que incluye todas las formas de fósforo en el agua.
20. fosf ortofos mg l: Concentración de ortofosfatos en miligramos por litro (mg/L), que es un nutriente importante.
21. dbo mg l: Demanda biológica de oxígeno en miligramos por litro (mg/L), que mide la cantidad de oxígeno requerido por microorganismos para descomponer materia orgánica.
22. dqo mg l: Demanda química de oxígeno en miligramos por litro (mg/L), que mide la cantidad total de oxígeno requerido
     para oxidar materia orgánica e inorgánica.
23. turbiedad ntu: Turbidez del agua medida en unidades NTU (Nephelometric Turbidity Units), que indica la claridad del agua.
24. hidr deriv petr ug l: Hidrocarburos derivados del petróleo en microgramos por litro (µg/L), que indican contaminación por productos petroleros.
25. cr total mg l: Concentración total de cromo en miligramos por litro (mg/L), un metal pesado que puede ser tóxico.
26. cd total mg l: Concentración total de cadmio en miligramos por litro (mg/L), otro metal pesado que es tóxico en altas concentraciones.
27. clorofila a ug l: Concentración de clorofila a en microgramos por litro (µg/L), que indica la cantidad de fitoplancton en el agua.
28. microcistina ug l: Concentración de microcistinas en microgramos por litro (µg/L), que son toxinas producidas por ciertas algas.
29. ica: Índice de calidad del agua, que puede ser un valor calculado para evaluar la calidad general del agua.
30. calidad de agua: Clasificación general de la calidad del agua basada en los parámetros medidos.

In [116]:
import pandas as pd

# Leer el archivo csv original especificando el delimitador como ';'
df_original = pd.read_csv("Calidad_de_agua_2021.csv", sep=';')

# Creamos una copia del DataFrame
df = df_original.copy()

In [117]:
df

orden                                     sitios   codigo       fecha  \
0        1               Canal Villanueva y Río Luján    TI001   24/2/2021   
1        1               Canal Villanueva y Río Luján    TI001   19/5/2021   
2        1               Canal Villanueva y Río Luján    TI001   25/8/2021   
3        1               Canal Villanueva y Río Luján    TI001  27/10/2021   
4        2               Río Lujan y Arroyo Caraguatá    TI006   24/2/2021   
5        2               Río Lujan y Arroyo Caraguatá    TI006   19/5/2021   
6        2               Río Lujan y Arroyo Caraguatá    TI006   25/8/2021   
7        2               Río Lujan y Arroyo Caraguatá    TI006  27/10/2021   
8        3                Canal Aliviador y Río Lujan    TI002   24/2/2021   
9        3                Canal Aliviador y Río Lujan    TI002   19/5/2021   
10       3                Canal Aliviador y Río Lujan    TI002   25/8/2021   
11       3                Canal Aliviador y Río Lujan    TI002  27/10/2021   
12       4      Río Carapachay y Arroyo Gallo Fiambre    TI003   24/2/2021   
13       4      Río Carapachay y Arroyo Gallo Fiambre    TI003   19/5/2021   
14       4      Río Carapachay y Arroyo Gallo Fiambre    TI003   25/8/2021   
15       4      Río Carapachay y Arroyo Gallo Fiambre    TI003  27/10/2021   
16       5                Río Reconquista y Río Lujan    TI004   24/2/2021   
17       5                Río Reconquista y Río Lujan    TI004   19/5/2021   
18       5                Río Reconquista y Río Lujan    TI004   25/8/2021   
19       5                Río Reconquista y Río Lujan    TI004  27/10/2021   
20       6         Rio Tigre 100m antes del Rio Luján    TI005   24/2/2021   
21       6         Rio Tigre 100m antes del Rio Luján    TI005   19/5/2021   
22       6         Rio Tigre 100m antes del Rio Luján    TI005   25/8/2021   
23       6         Rio Tigre 100m antes del Rio Luján    TI005  27/10/2021   
24       7             Río Lujan y Canal San Fernando    TI007   24/2/2021   
25       7             Río Lujan y Canal San Fernando    TI007   19/5/2021   
26       7             Río Lujan y Canal San Fernando    TI007   25/8/2021   
27       7             Río Lujan y Canal San Fernando    TI007  27/10/2021   
28       8              Río Capitán y Río San Antonio    TI008   24/2/2021   
29       8              Río Capitán y Río San Antonio    TI008   19/5/2021   
30       8              Río Capitán y Río San Antonio    TI008   25/8/2021   
31       8              Río Capitán y Río San Antonio    TI008  27/10/2021   
32       9             Arroyo Abra Vieja y Santa Rosa    TI009   24/2/2021   
33       9             Arroyo Abra Vieja y Santa Rosa    TI009   19/5/2021   
34       9             Arroyo Abra Vieja y Santa Rosa    TI009   25/8/2021   
35       9             Arroyo Abra Vieja y Santa Rosa    TI009  27/10/2021   
36      10                                   Del Arca    SF015   24/2/2021   
37      10                                   Del Arca    SF015   19/5/2021   
38      10                                   Del Arca    SF015   25/8/2021   
39      10                                   Del Arca    SF015  27/10/2021   
40      11                          Espigón La Farola    SI021   24/2/2021   
41      11                          Espigón La Farola    SI021   19/5/2021   
42      11                          Espigón La Farola    SI021   25/8/2021   
43      11                          Espigón La Farola    SI021  27/10/2021   
44      12                          Reserva Ecológica    SI022   24/2/2021   
45      12                          Reserva Ecológica    SI022   19/5/2021   
46      12                          Reserva Ecológica    SI022   25/8/2021   
47      12                          Reserva Ecológica    SI022  27/10/2021   
48      13                   Playa Espigón de Pacheco    SI024   24/2/2021   
49      13                   Playa Espigón de Pacheco    SI024   19/5/2021   
50      13                 

In [118]:
#Aca esto es para que solo nos muestre todas las columnas
pd.set_option('display.max_columns', None)

# Aca miramos unas filas raras  
x1 = df.iloc[163:165]

In [119]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   orden                    166 non-null    int64 
 1   sitios                   166 non-null    object
 2   codigo                   166 non-null    object
 3   fecha                    166 non-null    object
 4   año                      166 non-null    int64 
 5   campaña                  166 non-null    object
 6   tem_agua                 166 non-null    object
 7   tem_aire                 162 non-null    object
 8   od                       166 non-null    object
 9   ph                       166 non-null    object
 10  olores                   166 non-null    object
 11  color                    166 non-null    object
 12  espumas                  166 non-null    object
 13  mat_susp                 166 non-null    object
 14  colif_fecales_ufc_100ml  166 non-null    o

In [120]:
df.value_counts()

orden  sitios                                     codigo   fecha       año   campaña    tem_agua     tem_aire     od           ph           olores       color        espumas      mat_susp     colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  nitrato_mg_l  nh4_mg_l     p_total_l_mg_l  fosf_ortofos_mg_l  dbo_mg_l     dqo_mg_l     turbiedad_ntu  hidr_deriv_petr_ug_l  cr_total_mg_l  cd_total_mg_l  clorofila_a_ug_l  microcistina_ug_l  ica          calidad_de_agua           
1      Canal Villanueva y Río Luján               TI001    19/5/2021   2021  otoño      15.6         13           8.28         6.79         Ausencia     Ausencia     Ausencia     Ausencia     4000                     200                    50                 3             0.98         0.2             0.2                4.1          <2.0         <30            110                   <100           <0.005         <0.001            < 10.000           48           Muy deteriorada               1
             

In [121]:
df['espumas'].value_counts(dropna=False)

espumas
Ausencia       138
no se midió     15
Presencia        8
Ausencia         5
Name: count, dtype: int64

In [122]:
# Aca identificamos las que estan escritas dif (para saber sus diferencias tambien)
for idx in df["espumas"].value_counts().items():
    print(idx)



('Ausencia', 138)
('no se midió', 15)
('Presencia', 8)
('Ausencia ', 5)


In [123]:

# Remplazo los valores q estan escritos distintos para q esten iguales
df['espumas'] = df['espumas'].replace("Ausencia ", 'Ausencia')
df['espumas'] = df['espumas'].replace("no se midió", pd.NA)


In [124]:
# Aca identificamos las la dif entre como estan escritos
for idx in df["espumas"].value_counts().items():
    print(idx)

df["espumas"].isna().sum()
print(df["espumas"].unique())


('Ausencia', 143)
('Presencia', 8)
['Ausencia' 'Presencia' <NA>]


In [125]:
print(df["espumas"].isna().sum())

15


In [126]:
#Los pasamos a Booleano para mejor manupilacion 
df['espumas'] = df['espumas'].map({'Presencia': True, 'Ausencia': False}, na_action= 'ignore')

#ESTO NO ES BUENA IDEA USARLO porque sino te elimina los nulos
#df['espumas'] = df['espumas'].astype(bool)

In [127]:
df["espumas"].isna().sum()

np.int64(15)

In [128]:
#Empezamos a curar otra columna, este caso "fecha"
df['fecha'].value_counts(dropna=False)

fecha
27/10/2021    42
25/8/2021     42
19/5/2021     41
24/2/2021     41
Name: count, dtype: int64

In [129]:
#pasamos la fecha a formato date

df['fecha'] = pd.to_datetime(df['fecha'], format='%d/%m/%Y')

In [130]:
# me dice cuantos sitios hay ----Esto es para que este en odren alfabetico
fre = df['sitios'].value_counts(dropna=False).sort_index()

fre

sitios
Arroyo Abra Vieja y Santa Rosa               4
Arroyo El Gato                               4
Arroyo Santo Domingo                         4
Arroyo Sarandí                               4
Balneario Municipal                          4
Balneario Palo Blanco                        4
Boca Cerrada (Res.Nat. Punta Lara)           1
Calle 14 y Costa - Salida cloaca             4
Camping Eva Perón                            4
Canal Aliviador y Río Lujan                  4
Canal Villanueva y Río Luján                 4
Club de Pescadores                           4
Costa de Villa Domínico                      4
Costa y Melo                                 4
Costanera Hudson Calle 63                    4
Costanera Norte - Espigón Abanico            4
Del Arca                                     4
Diagonal 66 (descarga cloaca)                4
Ensenada Prefectura Isla Santiago            4
Escollera de Propaneros                      4
Espigón La Farola                            4
Espora

In [131]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   orden                    166 non-null    int64         
 1   sitios                   166 non-null    object        
 2   codigo                   166 non-null    object        
 3   fecha                    166 non-null    datetime64[ns]
 4   año                      166 non-null    int64         
 5   campaña                  166 non-null    object        
 6   tem_agua                 166 non-null    object        
 7   tem_aire                 162 non-null    object        
 8   od                       166 non-null    object        
 9   ph                       166 non-null    object        
 10  olores                   166 non-null    object        
 11  color                    166 non-null    object        
 12  espumas                  151 non-nul

In [132]:
fre = df['codigo'].value_counts(dropna=False).sort_index()
fre
# RARI la que pone EN-adic
df["codigo"].isna().sum()

np.int64(0)

In [133]:
fre = df['campaña'].value_counts()
fre
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   orden                    166 non-null    int64         
 1   sitios                   166 non-null    object        
 2   codigo                   166 non-null    object        
 3   fecha                    166 non-null    datetime64[ns]
 4   año                      166 non-null    int64         
 5   campaña                  166 non-null    object        
 6   tem_agua                 166 non-null    object        
 7   tem_aire                 162 non-null    object        
 8   od                       166 non-null    object        
 9   ph                       166 non-null    object        
 10  olores                   166 non-null    object        
 11  color                    166 non-null    object        
 12  espumas                  151 non-nul

In [134]:
fre =  df['tem_agua'].value_counts(dropna=False)
fre
#df["tem_agua"].isna().sum()

tem_agua
no se midió    17
24.7            6
14.8            5
26              5
14.5            5
14.4            4
25.5            4
11              4
10              4
15              3
23.8            3
24              3
14.7            3
15.7            3
24.9            3
13              3
26.1            3
16              2
25.7            2
25.6            2
24.6            2
15.1            2
23.5            2
22.4            2
14.6            2
18              2
13.8            2
15.2            2
24.2            2
24.8            2
14              2
21.4            2
16.7            2
12              2
7               2
20              2
21              2
12.9            2
16.2            2
25              2
15.3            2
27              2
22              2
24.5            2
15.6            1
24.4            1
25.4            1
25.2            1
25.8            1
13.5            1
17              1
23.6            1
27.5            1
28              1
15.9            1
2

In [135]:
#tenes 2 formas de hacer esto 1, reemplazas a los "no se midio" por NaN y dsp lo pasa a tipo numerico, en este caso uso el ignore porque queiro 
# asegurarme que unicamente los "no se midio" se esten cambiando
df['tem_agua'] = df['tem_agua'].replace("no se midió", pd.NA)
df['tem_agua'] = pd.to_numeric(df['tem_agua'] )# , errors='ignore')

# O pones solo esto que reemplaza a lo que es un numero, a nulo, coerce fuerza a los valores que no sean numeros a convertirse en nulos 

#df['tem_agua'] = pd.to_numeric(df['tem_agua'], errors='coerce')

In [136]:
df['tem_aire'] = df['tem_aire'].replace("no se midió", pd.NA)
df['tem_aire'] = pd.to_numeric(df['tem_aire']) #, errors='ignore')
fre =  df['tem_aire'].value_counts(dropna=False)
fre

tem_aire
NaN     29
28.0    12
26.0    11
27.0    10
11.0    10
12.0     7
13.0     7
14.5     6
8.0      5
32.0     5
19.0     5
12.8     4
20.0     4
18.0     4
4.0      4
29.1     3
27.5     3
7.0      3
14.0     3
16.0     3
10.0     2
29.0     2
24.0     2
17.0     2
24.2     2
22.0     2
13.8     2
9.0      2
31.0     2
33.0     1
25.0     1
32.5     1
13.5     1
24.6     1
13.4     1
12.6     1
15.0     1
28.5     1
12.5     1
Name: count, dtype: int64

In [137]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   orden                    166 non-null    int64         
 1   sitios                   166 non-null    object        
 2   codigo                   166 non-null    object        
 3   fecha                    166 non-null    datetime64[ns]
 4   año                      166 non-null    int64         
 5   campaña                  166 non-null    object        
 6   tem_agua                 149 non-null    float64       
 7   tem_aire                 137 non-null    float64       
 8   od                       166 non-null    object        
 9   ph                       166 non-null    object        
 10  olores                   166 non-null    object        
 11  color                    166 non-null    object        
 12  espumas                  151 non-nul

In [138]:
pd.set_option('display.max_rows', None)


In [139]:
fre =  df['ica'].value_counts(dropna=False)
fre

ica
no se midió    15
46             11
49             10
39              9
42              9
38              8
41              8
43              7
34              6
48              5
45              5
44              5
37              4
58              4
36              4
35              4
40              4
26              3
50              3
47              3
28              3
52              3
33              3
60              3
51              3
53              2
64              2
65              2
54              2
57              2
30              2
55              1
29              1
31              1
75              1
80              1
59              1
61              1
56              1
72              1
67              1
74              1
62              1
Name: count, dtype: int64

In [140]:
df['od'] = df['od'].replace("no se midió", pd.NA)
df['od'] = pd.to_numeric(df['od'], errors='ignore')

df['ph'] = df['ph'].replace("no se midió", pd.NA)
df['ph'] = pd.to_numeric(df['ph'], errors='ignore')

df['colif_fecales_ufc_100ml'] = df['colif_fecales_ufc_100ml'].replace("no se midió", pd.NA) 
df['colif_fecales_ufc_100ml'] = pd.to_numeric(df['colif_fecales_ufc_100ml'], errors='ignore')

df['escher_coli_ufc_100ml'] = df['escher_coli_ufc_100ml'].replace("no se midió", pd.NA)
df['escher_coli_ufc_100ml'] = pd.to_numeric(df['escher_coli_ufc_100ml'], errors='ignore')

df['enteroc_ufc_100ml'] = df['enteroc_ufc_100ml'].replace("no se midió", pd.NA)
df['enteroc_ufc_100ml'] = pd.to_numeric(df['enteroc_ufc_100ml'], errors='ignore')

df['nitrato_mg_l'] = df['nitrato_mg_l'].replace("no se midió", pd.NA)
df['nitrato_mg_l'] = pd.to_numeric(df['nitrato_mg_l'], errors='ignore')

df['nh4_mg_l'] = df['nh4_mg_l'].replace("no se midió", pd.NA)
df['nh4_mg_l'] = pd.to_numeric(df['nh4_mg_l'], errors='ignore')

df['p_total_l_mg_l'] = df['p_total_l_mg_l'].replace("no se midió", pd.NA)
df['p_total_l_mg_l'] = pd.to_numeric(df['p_total_l_mg_l'], errors='ignore')

df['fosf_ortofos_mg_l'] = df['fosf_ortofos_mg_l'].replace("no se midió", pd.NA)
df['fosf_ortofos_mg_l'] = pd.to_numeric(df['fosf_ortofos_mg_l'], errors='ignore')

df['dbo_mg_l'] = df['dbo_mg_l'].replace("no se midió", pd.NA)
df['dbo_mg_l'] = pd.to_numeric(df['dbo_mg_l'], errors='ignore')

df['dqo_mg_l'] = df['dqo_mg_l'].replace("no se midió", pd.NA)
df['dqo_mg_l'] = pd.to_numeric(df['dqo_mg_l'], errors='ignore')

df['turbiedad_ntu'] = df['turbiedad_ntu'].replace("no se midió", pd.NA)
df['turbiedad_ntu'] = pd.to_numeric(df['turbiedad_ntu'], errors='ignore')

df['hidr_deriv_petr_ug_l'] = df['hidr_deriv_petr_ug_l'].replace("no se midió", pd.NA)
df['hidr_deriv_petr_ug_l'] = pd.to_numeric(df['hidr_deriv_petr_ug_l'], errors='ignore')

df['cr_total_mg_l'] = df['cr_total_mg_l'].replace("no se midió", pd.NA)
df['cr_total_mg_l'] = pd.to_numeric(df['cr_total_mg_l'], errors='ignore')

df['cd_total_mg_l'] = df['cd_total_mg_l'].replace("no se midió", pd.NA)
df['cd_total_mg_l'] = pd.to_numeric(df['cd_total_mg_l'], errors='ignore')

df['clorofila_a_ug_l'] = df['clorofila_a_ug_l'].replace("no se midió", pd.NA)
df['clorofila_a_ug_l'] = pd.to_numeric(df['clorofila_a_ug_l'], errors='ignore')

df['microcistina_ug_l'] = df['microcistina_ug_l'].replace("no se midió", pd.NA)
df['microcistina_ug_l'] = pd.to_numeric(df['microcistina_ug_l'], errors='ignore')

df['ica'] = df['ica'].replace("no se midió", pd.NA)
df['ica'] = pd.to_numeric(df['ica'], errors='ignore')


C:\Users\berni\AppData\Local\Temp\ipykernel_26080\1205627107.py:2: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['od'] = pd.to_numeric(df['od'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_26080\1205627107.py:5: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['ph'] = pd.to_numeric(df['ph'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_26080\1205627107.py:8: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
  df['colif_fecales_ufc_100ml'] = pd.to_numeric(df['colif_fecales_ufc_100ml'], errors='ignore')
C:\Users\berni\AppData\Local\Temp\ipykernel_26080\1205627107.py:11: FutureWarning: errors='ignore' is deprecated

In [141]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 166 entries, 0 to 165
Data columns (total 31 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   orden                    166 non-null    int64         
 1   sitios                   166 non-null    object        
 2   codigo                   166 non-null    object        
 3   fecha                    166 non-null    datetime64[ns]
 4   año                      166 non-null    int64         
 5   campaña                  166 non-null    object        
 6   tem_agua                 149 non-null    float64       
 7   tem_aire                 137 non-null    float64       
 8   od                       150 non-null    float64       
 9   ph                       150 non-null    float64       
 10  olores                   166 non-null    object        
 11  color                    166 non-null    object        
 12  espumas                  151 non-nul

In [142]:
#Los pasamos a Booleano para mejor manupilacion 

# Aca identificamos las la dif entre como estan escritos
for idx in df['olores'].value_counts().items():
    print(idx)

# Remplazo los valores q estan escritos distintos para q esten iguales
df['olores'] = df['olores'].replace("Ausencia ", 'Ausencia')
df['olores'] = df['olores'].replace("no se midió", pd.NA)

for idx in df['olores'].value_counts().items():
    print(idx)

#Los pasamos a Booleano para mejor manupilacion 
df['olores'] = df['olores'].map({'Presencia': True, 'Ausencia': False})

#df['olores'] = df['olores'].map({'Presencia': True, 'Ausencia': False, 'Ausencia ': False, 'no se midió': pd.NA})
#df['olores'] = df['olores'].astype(bool)
#df['olores'].value_counts()

('Ausencia', 126)
('Presencia', 19)
('no se midió', 15)
('Ausencia ', 6)
('Ausencia', 132)
('Presencia', 19)


In [143]:
df['olores'].value_counts(dropna=False)


olores
False    132
True      19
NaN       15
Name: count, dtype: int64

In [144]:
df['olores']

0      False
1      False
2      False
3      False
4      False
5      False
6      False
7      False
8       True
9      False
10      True
11      True
12     False
13     False
14     False
15     False
16      True
17      True
18      True
19      True
20      True
21      True
22      True
23      True
24      True
25      True
26      True
27     False
28     False
29     False
30     False
31     False
32     False
33     False
34     False
35     False
36     False
37       NaN
38     False
39     False
40       NaN
41       NaN
42       NaN
43       NaN
44     False
45     False
46     False
47     False
48     False
49     False
50     False
51     False
52     False
53     False
54     False
55     False
56     False
57     False
58     False
59     False
60     False
61     False
62     False
63     False
64     False
65     False
66     False
67     False
68     False
69       NaN
70     False
71     False
72     False
73       NaN
74     False
75     False
76     False

In [145]:
df['calidad_de_agua'] = df['calidad_de_agua'].replace("no se midió", pd.NA)
df['calidad_de_agua'].value_counts(dropna=False)


calidad_de_agua
Extremadamente deteriorada    81
Muy deteriorada               63
<NA>                          15
Deteriorada                    6
Levemente deteriorada          1
Name: count, dtype: int64

In [146]:
# Aca identificamos las la dif entre como estan escritos
for idx in df['color'].value_counts().items():
    print(idx)

# Remplazo los valores q estan escritos distintos para q esten iguales
df['color'] = df['color'].replace("Ausencia ", 'Ausencia')
df['color'] = df['color'].replace("ausenca", 'Ausencia')
df['color'] = df['color'].replace("no se midió", pd.NA)


('Ausencia', 127)
('Presencia', 18)
('no se midió', 15)
('Ausencia ', 5)
('ausenca', 1)


In [147]:

#Los pasamos a Booleano para mejor manupilacion 
df['color'] = df['color'].map({'Presencia': True, 'Ausencia': False})

In [148]:
# Aca identificamos las la dif entre como estan escritos
for idx in df['mat_susp'].value_counts().items():
    print(idx)

# Remplazo los valores q estan escritos distintos para q esten iguales
df['mat_susp'] = df['mat_susp'].replace("Ausencia ", 'Ausencia')
df['mat_susp'] = df['mat_susp'].replace("no se midió", pd.NA)

('Ausencia', 116)
('Presencia', 30)
('no se midió', 15)
('Ausencia ', 5)


In [149]:

#Los pasamos a Booleano para mejor manupilacion 
df['mat_susp'] = df['mat_susp'].map({'Presencia': True, 'Ausencia': False})

In [150]:
df.value_counts(dropna=False)

orden  sitios                                     codigo   fecha       año   campaña    tem_agua  tem_aire  od     ph    olores  color  espumas  mat_susp  colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  nitrato_mg_l  nh4_mg_l  p_total_l_mg_l  fosf_ortofos_mg_l  dbo_mg_l  dqo_mg_l  turbiedad_ntu  hidr_deriv_petr_ug_l  cr_total_mg_l  cd_total_mg_l  clorofila_a_ug_l  microcistina_ug_l  ica   calidad_de_agua           
1      Canal Villanueva y Río Luján               TI001    2021-02-24  2021  verano     25.6      27.0      3.91   6.96  False   False  False    True      4000                     1000.0                 330.0              6.5           0.4       0.92            0.37               <2.0      <30       60             <100                  <5.0           <0.001         0                 <0.15              42.0  Extremadamente deteriorada    1
                                                           2021-05-19  2021  otoño      15.6      13.0      8.28   6.79

In [151]:
df_estaciones = df.query("campaña == 'otoño'")
#df_estaciones = df_estaciones.query()
df_estaciones["colif_fecales_ufc_100ml"].isna().sum()

np.int64(12)

In [152]:
# Como no hay datos de Espigon La Farola, (una reserva natural)
# Decidimos sacarlo 

# Filtrar las filas donde 'sitios' no sea 'Espigón La Farola'
df_sin_espigon = df[df['sitios'] != 'Espigón La Farola'].index

df_copy = df.copy()

# Usar drop para eliminar esas filas basadas en los índices
df_copy = df_copy.loc[df['sitios'] != 'Espigón La Farola']

df_copy.value_counts(dropna=False)




orden  sitios                                     codigo   fecha       año   campaña    tem_agua  tem_aire  od     ph    olores  color  espumas  mat_susp  colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  nitrato_mg_l  nh4_mg_l  p_total_l_mg_l  fosf_ortofos_mg_l  dbo_mg_l  dqo_mg_l  turbiedad_ntu  hidr_deriv_petr_ug_l  cr_total_mg_l  cd_total_mg_l  clorofila_a_ug_l  microcistina_ug_l  ica   calidad_de_agua           
1      Canal Villanueva y Río Luján               TI001    2021-02-24  2021  verano     25.6      27.0      3.91   6.96  False   False  False    True      4000                     1000.0                 330.0              6.5           0.4       0.92            0.37               <2.0      <30       60             <100                  <5.0           <0.001         0                 <0.15              42.0  Extremadamente deteriorada    1
                                                           2021-05-19  2021  otoño      15.6      13.0      8.28   6.79

In [153]:
# Filtrar las filas donde los primeros 6 valores no sean NaN
filas_parcialmente_nan = df_copy[df_copy.isna().sum(axis=1) >= len(df_copy.columns) - 6]


In [154]:
# Mostrar las filas que cumplen con la condición
filas_parcialmente_nan.shape

(11, 31)

In [155]:
# inplace=True: Asegura que las filas se eliminen directamente del DataFrame, sin necesidad de crear una copia.

indices_a_eliminar = filas_parcialmente_nan.index

df_copy.drop(indices_a_eliminar, inplace=True)


In [156]:
df_copy = df_copy.drop('orden', axis=1)

In [157]:
conteo_combinado = df_copy.groupby(['sitios','codigo']).size()

conteo_combinado = conteo_combinado.sort_index()
conteo_combinado

sitios                                     codigo 
Arroyo Abra Vieja y Santa Rosa             TI009      4
Arroyo El Gato                             EN083      4
Arroyo Santo Domingo                       AV053      4
Arroyo Sarandí                             AV052      4
Balneario Municipal                        BS094      3
Balneario Palo Blanco                      BS092      3
Boca Cerrada (Res.Nat. Punta Lara)         EN-adic    1
Calle 14 y Costa - Salida cloaca           BZ078      4
Camping Eva Perón                          EN081      3
Canal Aliviador y Río Lujan                TI002      4
Canal Villanueva y Río Luján               TI001      4
Club de Pescadores                         CA046      3
Costa de Villa Domínico                    AV055      4
Costa y Melo                               VL031      4
Costanera Hudson Calle 63                  BZ080      4
Costanera Norte - Espigón Abanico          CA044      3
Del Arca                                   SF015     

In [158]:
df.value_counts(dropna=False).sum()

np.int64(166)

In [159]:
df_copy.value_counts(dropna=False)

sitios                                     codigo   fecha       año   campaña    tem_agua  tem_aire  od     ph    olores  color  espumas  mat_susp  colif_fecales_ufc_100ml  escher_coli_ufc_100ml  enteroc_ufc_100ml  nitrato_mg_l  nh4_mg_l  p_total_l_mg_l  fosf_ortofos_mg_l  dbo_mg_l  dqo_mg_l  turbiedad_ntu  hidr_deriv_petr_ug_l  cr_total_mg_l  cd_total_mg_l  clorofila_a_ug_l  microcistina_ug_l  ica   calidad_de_agua           
Arroyo Abra Vieja y Santa Rosa             TI009    2021-02-24  2021  verano     24.7      28.0      4.17   6.85  False   False  False    False     1900                     360.0                  150.0              2.5           0.34      0.59            0.3                <2.0      <30       50             <100                  <5.0           <0.001         0.009             <0.15              49.0  Muy deteriorada               1
                                                    2021-05-19  2021  otoño      12.8      8.0       8.20   6.89  False   False  Fals

In [160]:
conteo_combinado = df_copy.groupby(['sitios','codigo']).size()

conteo_combinado = conteo_combinado.sort_index()
conteo_combinado

sitios                                     codigo 
Arroyo Abra Vieja y Santa Rosa             TI009      4
Arroyo El Gato                             EN083      4
Arroyo Santo Domingo                       AV053      4
Arroyo Sarandí                             AV052      4
Balneario Municipal                        BS094      3
Balneario Palo Blanco                      BS092      3
Boca Cerrada (Res.Nat. Punta Lara)         EN-adic    1
Calle 14 y Costa - Salida cloaca           BZ078      4
Camping Eva Perón                          EN081      3
Canal Aliviador y Río Lujan                TI002      4
Canal Villanueva y Río Luján               TI001      4
Club de Pescadores                         CA046      3
Costa de Villa Domínico                    AV055      4
Costa y Melo                               VL031      4
Costanera Hudson Calle 63                  BZ080      4
Costanera Norte - Espigón Abanico          CA044      3
Del Arca                                   SF015     

datos a tener en cuenta: -- esto esta sacado del folleto de calidad de agua, Resolución ACUMAR 46/2017 y la Resolución ADA 42
---------????- hay valores que se mostraron sin un valor especifico, podriamos ver si tiene correlacion,
fosfatos y DQO

.El oxigeno disuelto -- lo normal es que este en valores superiores a 5mg/l
.El ph -- lo normal es que este entre 6,5 y 9 (aunque podriamos mostrar como zonas delicadas los extremos
.La temperatura que este por debajo de los 35 
.La Turbiez no puede superar el 100 
.Color, Olor o cosas visibles flotando no tendria que haber
.coliformes fecales hasta 150 ufc/100ml
.esche hasta 126
.enterococos hasta 33
.nitratos hasta 125 mg/l
.amonio hasta 0,5
.fosforo hasta 0,025
.clorofila 50
.microcistina 10
.dbo 10
. hidrocarburos <50
.cromo 50
.cadmio 5

In [161]:
conteo_combinado = df_copy.groupby(['sitios','codigo']).size()

conteo_combinado = conteo_combinado.sort_index()

#df_copy["sitios"].value_counts()
#print(conteo_combinado)


In [162]:
df_copy.describe()

fecha     año    tem_agua    tem_aire  \
count                            151   151.0  149.000000  137.000000   
mean   2021-07-05 10:10:19.867549696  2021.0   19.165772   19.194161   
min              2021-02-24 00:00:00  2021.0    7.000000    4.000000   
25%              2021-02-24 00:00:00  2021.0   14.500000   12.600000   
50%              2021-08-25 00:00:00  2021.0   18.000000   18.000000   
75%              2021-10-27 00:00:00  2021.0   24.700000   27.000000   
max              2021-10-27 00:00:00  2021.0   28.000000   33.000000   
std                              NaN     0.0    5.675621    8.044781   

               od          ph  escher_coli_ufc_100ml  enteroc_ufc_100ml  \
count  150.000000  150.000000             148.000000         148.000000   
mean     6.645200    7.582067            5197.263514         927.337838   
min      0.590000    6.610000               5.000000           2.000000   
25%      4.242500    6.977500             297.500000          63.750000   
50%      6.260000    7.475000            1260.000000         250.000000   
75%      8.797500    7.857500            4100.000000         800.000000   
max     15.200000    9.660000           70000.000000       28000.000000   
std      3.107303    0.731462           11284.350013        2571.965736   

              ica  
count  151.000000  
mean    44.894040  
min     26.000000  
25%     38.000000  
50%     43.000000  
75%     49.000000  
max     80.000000  
std     10.065222

In [163]:
#Con respecto a los datos duplicados 2 decisiones posibles, dado a que habla de calidad de agua (personalmente prefiero) decantar por el peor valor, o hacer un promedio entre esos 2 valores
# ademas se podria percibir una disminucion de invierno a primavera (tomando el valor mas alto), es especulativo pero puede ser que justo vieron que se arrojo algun desecho

#PANDEMIA -- Probablemente no 
#para rellenar datos - buscar lugares cercanos geograficamente -- usar valores de las otras estaciones
